File: python/Gate Seven/number_guessing_app.py
import random

def random_number_generator():
    random_number = random.randint(1, 100)
    return random_number

def track_and_display_rating(attempts):
    if attempts == 1:
        return "Legendary"
    elif attempts == 2:
        return "Excellent"
    elif attempts == 3 or attempts == 4:
        return "Good"
    elif attempts == 5:
        return "Close!"
    else:
        return "Better luck"

def number_guessing_game(random_number):
    attempts = 1
    
    while (attempts <= 5):
        user_input = input("Guess the random number: ")
       
        if (input_validator(user_input) == True):
            user_input = int(user_input)
            if user_input != random_number:
                if user_input < random_number:
                    print ("Guess too low")
                if user_input > random_number:
                    print ("Guess too high")
                attempts += 1
            else:
                print("Correct!")
                break
        else: 
             print(f"Invalid input, you are on attempt {attempts + 1}")
        
    return attempts


def input_validator(user_input: str):
    is_valid = False
    if str(user_input).isdigit():
        if int(user_input) >= 1 and int(user_input) <= 100:
            is_valid = True 
        else:
            is_valid = False
            
    else:
        is_valid = False
    return is_valid
    

def deisplay_summary(random_number, attempts):
    correct_number = random_number
    rating = track_and_display_rating(attempts)
    
    return f"The Correct number is {correct_number}, you took {attempts} attempts, {rating}"

File: java/Gate Seven/NumberGuessingApp.java
import java.util.Scanner; 
import java.util.Random;

public class NumberGuessingApp{

	public static int randomNumberGenerator(){
		Random random = new Random();
		return (random.nextInt(1, 101));
	}

	public static String trackAndDisplayRating(int attempts){
		if (attempts == 1){
			return "Legendary";
		}
		else if (attempts == 2){
		  	return "Excellent";
		}
		else if (attempts == 3 || attempts == 4){
		  	return "Good";
		}
		else if (attempts == 5){
		  	return "Close!";
		}
		else {
		  	return "Better luck";
		}
	}

	public static boolean inputValidator(Scanner userInput){
		
        	boolean isValid = true;
        	while (isValid) { 
            		if (userInput.hasNextInt()) {
                		int value = userInput.nextInt();
            
                		if (value >= 1 && value <= 100) {
					isValid = true;
					return isValid;
                		} 
				else {
                    			System.out.print("Out of range, Try again: ");
					isValid = false;
                		}
            		} 	
			else {
				isValid = false;
                		System.out.print("Invalid input. Try again: ");
                		userInput.next();
           		}
        	}
        	return isValid;
	}

	public static void main (String [] args){
		Scanner userInput = new Scanner(System.in);
		System.out.print(inputValidator(userInput));
	}
} 